Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/Message/MessageReport.hs

```diff
@@ -11,20 +11,68 @@
 
  the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
 -}
+{-# LANGUAGE DerivingStrategies #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
 
 module Domain.Types.Message.MessageReport where
 
+import qualified Data.Aeson as A
+import Data.ByteString
 import qualified Data.Map as M
 import Data.OpenApi
+import qualified Database.Beam as B
+import Database.Beam.Backend
+import Database.Beam.Postgres
+import Database.PostgreSQL.Simple.FromField (FromField (fromField))
+import qualified Database.PostgreSQL.Simple.FromField as DPSF
 import qualified Domain.Types.Message.Message as Msg
 import Domain.Types.Person (Driver)
 import Kernel.Prelude
+import Kernel.Types.Common (fromFieldEnum)
 import Kernel.Types.Id
 
 type MessageDynamicFieldsType = M.Map Text Text
 
+fromFieldmessageDynamicFields ::
+  DPSF.Field ->
+  Maybe ByteString ->
+  DPSF.Conversion MessageDynamicFieldsType
+fromFieldmessageDynamicFields f mbValue = do
+  value' <- fromField f mbValue
+  case A.fromJSON value' of
+    A.Success a -> pure a
+    _ -> DPSF.returnError DPSF.ConversionFailed f "Conversion failed"
+
+instance FromField MessageDynamicFieldsType where
+  fromField = fromFieldmessageDynamicFields
+
+instance HasSqlValueSyntax be A.Value => HasSqlValueSyntax be MessageDynamicFieldsType where
+  sqlValueSyntax = sqlValueSyntax . A.toJSON
+
+instance BeamSqlBackend be => B.HasSqlEqualityCheck be MessageDynamicFieldsType
+
+instance FromField (Text, Text) => FromBackendRow Postgres MessageDynamicFieldsType
```

Comment:
constraint is redundant
`instance FromBackendRow Postgres MessageDynamicFieldsType`